plugin/src/main/java/io/fabric8/maven/plugin/SetupMojo.java

```diff
@@ -0,0 +1,295 @@
+        boolean pluginInserted = fmpPlugin == null;
+        if (updateVersion || pluginInserted) {
+            if (useVersionProperty) {
+                Element documentElement = doc.getDocumentElement();
+                Element properties = firstChild(documentElement, "properties");
+                if (properties == null) {
+                    properties = addChildAfter(appendAfterLastElement(properties, doc.createTextNode("\n      ")), "properties");
+                }
+                if (Strings.isNullOrBlank(currentVersion)) {
+                    addChildAfter(appendAfterLastElement(properties, doc.createTextNode("\n    ")), FABRIC8_MAVEN_PLUGIN_VERSION_PROPERTY, latestVersion);
+                    updated = true;
+                } else if (!Objects.equals(currentVersion, latestVersion)) {
+                    Element propertyElement = DomHelper.firstChild(properties, FABRIC8_MAVEN_PLUGIN_VERSION_PROPERTY);
+                    if (propertyElement != null) {
+                        propertyElement.setTextContent(latestVersion);
+                        updated = true;
+                    }
+                }
+            } else {
+                versionExpression = latestVersion;
+            }
+        }
+
+        if (fmpPlugin == null) {
+            fmpPlugin = findOrAddPlugin(doc, PLUGIN_GROUPID, PLUGIN_ARTIFACTID, versionExpression, configuration);
+            updated = true;
+        } else {
+            if (updateVersion || pluginInserted) {
+                String version = DomHelper.firstChildTextContent(fmpPlugin, "version");
+                if (version == null || !version.equals(versionExpression)) {
+                    Element versionElement = DomHelper.firstChild(fmpPlugin, "version");
+                    if (versionElement == null) {
+                        Element artifactId = DomHelper.firstChild(fmpPlugin, "artifactId");
+                        Text textNode = doc.createTextNode("\n        ");
+                        if (artifactId != null) {
+                            addChildAfter(artifactId, textNode);
+                        } else {
+                            appendAfterLastElement(fmpPlugin, textNode);
+                        }
+                        addChildAfter(textNode, "version", versionExpression);
+                    } else {
+                        versionElement.setTextContent(versionExpression);
+                    }
+                    updated = true;
+                }
+            }
+            if (configuration != null) {
+                Element oldConfig = firstChild(fmpPlugin, "configuration");
+                DomHelper.detach(oldConfig);
+                fmpPlugin.appendChild(configuration);
+                if (oldConfig == null) {
+                    fmpPlugin.appendChild(doc.createTextNode("\n      "));
+                }
+            }
+        }
+        if (fmpPlugin != null) {
+            Element executions = firstChild(fmpPlugin, "executions");
```

Comment:
Should executions always be added with our default ? It could be that a user has a custom `<executions>` setting (e.g. binding our mojos to different phases). This would be unconditionally overwritten here.